### PR TITLE
New CI

### PR DESCRIPTION
**Pull request by marioortizmanero**
This PR includes the new CI suggested in #213. It's merged over #215 to avoid merge conflicts later on.

### CODE REVIEW

**Comment by marioortizmanero** [comment]
I've been taking a look at [Bevy's CI](https://github.com/bevyengine/bevy/blob/main/.github/workflows/ci.yml) and maybe we could copy some of their stuff? I'm taking a look right now.

**Comment by marioortizmanero** [comment]
OK so I've added a flow that checks for broken links in the docs as well. This PR fixes all of them except for those that are impossible to fix until everything's merged. For example I added links to the new examples like https://github.com/ramsayleung/rspotify/blob/master/examples/auth_code.rs, but they won't work until the rewrite hits the `master` branch (they work with https://github.com/ramsayleung/rspotify/blob/auth-rewrite-part3/examples/auth_code.rs for now)

As to having a flow for each crate I'm not sure if it's worth it. Specially because the macros crate doesn't have that many tests, for example. I think having a flow for each client might be enough for our case.

**Comment by marioortizmanero** [comment]
After a lot of debugging, turns out it's very complicated to have a clean run of `cargo-deadlinks` (i.e. without errors). For example, ours fails due to an external crate, https://github.com/tokio-rs/tracing/pull/1078.

Thus, I'm adding a `cargo-deadlinks` flow, but it will continue on error for now. This is what Bevy does as well, so it makes sense to me. It's still useful to check from time to time and it avoids unfixable errors.

This is ready for review, @ramsayleung

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,51 +27,41 @@ jobs:
...
-      - name: Run cargo build
-        uses: actions-rs/cargo@v1
-        with:
-          command: build
-          args: -p rspotify -p rspotify-http -p rspotify-model -p rspotify-macros --no-default-features --features=${{ matrix.client }} --target ${{ matrix.target }} 
+      - name: Install cargo-deadlinks
+        run: cargo install cargo-deadlinks
+
+      - name: Build and Check
+        run: RUSTDOCFLAGS='-D warnings' cargo doc --features=cli,env-file
```
Comment (ramsayleung) [inline .github/workflows/ci.yml:48]: Do we just check dead links for `cli` and `env-file` features, with ignoring other features?

**Comment by ramsayleung** [comment]
I have to say, the `cargo-deadlinks` is great, it helps us find the broken links, instead of identifying it manually.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (marioortizmanero) [reply .github/workflows/ci.yml:48]: The rest of the features don't add anything new to the docs.

**Review by ramsayleung — APPROVED** [review]

**Pull request merged**